lighthouse-core/lib/network-request.js

```diff
@@ -0,0 +1,244 @@
+
+    this._updateResponseReceivedTimeIfNecessary();
+  }
+
+  /**
+   * @param {LH.Crdp.Network.ResourceChangedPriorityEvent} data
+   */
+  onResourceChangedPriority(data) {
+    this.priority = () => data.newPriority;
+  }
+
+  /**
+   * @param {LH.Crdp.Network.RequestWillBeSentEvent} data
+   */
+  onRedirectResponse(data) {
+    if (!data.redirectResponse) throw new Error('Missing redirectResponse data');
+    this._onResponse(data.redirectResponse, data.timestamp, data.type);
+    this.finished = true;
+    this.endTime = data.timestamp;
+
+    this._updateResponseReceivedTimeIfNecessary();
+  }
+
+  /**
+   * @param {LH.Crdp.Network.Response} response
+   * @param {number} timestamp
+   * @param {LH.Crdp.Network.ResponseReceivedEvent['type']=} resourceType
+   */
+  _onResponse(response, timestamp, resourceType) {
+    this.connectionId = String(response.connectionId);
+    this.connectionReused = response.connectionReused;
+
+    if (response.protocol) this.protocol = response.protocol;
+
+    this._responseReceivedTime = timestamp;
+
+    this.transferSize = response.encodedDataLength;
+    this._transferSize = response.encodedDataLength;
+    if (typeof response.fromDiskCache === 'boolean') this._fromDiskCache = response.fromDiskCache;
+
+    this.statusCode = response.status;
+
+    this._timing = response.timing;
+    if (resourceType) this._resourceType = resourceTypes[resourceType];
+    this._mimeType = response.mimeType;
+    this._responseHeaders = NetworkRequest._headersMapToHeadersArray(response.headers);
+
+    this._fetchedViaServiceWorker = !!response.fromServiceWorker;
+
+    if (response.timing) this._recomputeTimesWithResourceTiming(response.timing);
+  }
+
+  /**
+   * @param {LH.Crdp.Network.ResourceTiming} timing
+   */
+  _recomputeTimesWithResourceTiming(timing) {
+    // Take startTime and responseReceivedTime from timing data for better accuracy.
+    // Timing's requestTime is a baseline in seconds, rest of the numbers there are ticks in millis.
+    this.startTime = timing.requestTime;
+    const headersReceivedTime = timing.requestTime + timing.receiveHeadersEnd / 1000.0;
+    if (!this._responseReceivedTime || this._responseReceivedTime < 0) {
+      this._responseReceivedTime = headersReceivedTime;
+    }
+
+    this._responseReceivedTime = Math.min(this._responseReceivedTime, headersReceivedTime);
+    this._responseReceivedTime = Math.max(this._responseReceivedTime, this.startTime);
+    this.endTime = Math.max(this.endTime, this._responseReceivedTime);
+  }
+
+  _updateResponseReceivedTimeIfNecessary() {
+    this._responseReceivedTime = Math.min(this.endTime, this._responseReceivedTime);
+  }
+
+  /**
+   * @param {LH.Crdp.Network.Headers} headersMap
+   * @return {Array<LH.WebInspector.HeaderValue>}
+   */
+  static _headersMapToHeadersArray(headersMap) {
+    const result = [];
+    for (const name of Object.keys(headersMap)) {
+      const values = headersMap[name].split('\n');
```

Comment:
which method in sdk.networkrequest is this disguised as? `_computeHeaderValue`

Reply:
Not even disguised verbatim s/in/of/ ;)
https://github.com/ChromeDevTools/devtools-frontend/blob/3415ee28e86a3f4bcc2e15b652d22069938df3a6/front_end/sdk/NetworkManager.js#L285-L297

added a comment